sympy/physics/units/quantities.py

```diff
@@ -209,3 +227,59 @@ def convert_to(self, other, unit_system="SI"):
     def free_symbols(self):
         """Return free symbols from quantity."""
         return set()
+
+class Offset(AtomicExpr):
+    """
+    Class to represent a temperature offset quantity as
+    the product of a numerical value and a temperature scale
+
+    Examples
+    ========
+
+    >>> from sympy.physics.units import oC, oF
+    >>> from sympy.physics.units import Offset
+    >>> Offset(0, oC)
+    Offset(0, degree_celsius)
+    >>> Offset(0, oC)(oF)
+    Offset(32, degree_fahrenheit)
+    >>> Offset(0, oC).doit().n()
+    273.15*kelvin
+    """
+
+    def __new__(cls, magnitude, units):
+
+        magnitude = sympify(magnitude)
+
+        if not isinstance(units, Quantity):
+            raise TypeError("Offset units must be a Quantity")
+
+        from sympy.physics.units import temperature
+        if units.dimension is not temperature:
+            raise ValueError("Offset units must be a temperature")
+
+        obj = AtomicExpr.__new__(cls, magnitude, units)
+        obj._magnitude = magnitude
+        obj._units = units
+        return obj
+
+    @property
+    def magnitude(self):
+        return self._magnitude
+
+    @property
+    def units(self):
+        return self._units
+
+    def _latex(self, printer):
+        return "%s %s" %(printer._print(self.magnitude), printer._print(self.units))
+
+    def __call__(self, other):
+        """Converts temperatures to different temperature scales"""
+        from .util import convert_temperature
+        return convert_temperature(self, other, unit_system="SI")
+
+    def doit(self):
```

Comment:
To keep this Henry useful I think we need to query for the quantity's offset and do the conversion to that...similar to get scale offset function below.

Reply:
In this case the offset of oC and oF would be Kelvin, right?  I want to make sure I understand what to query for correctly.